src/openrct2/object/ImageTable.cpp

```diff
@@ -486,7 +486,7 @@ std::vector<std::pair<std::string, Image>> ImageTable::GetImageSources(IReadObje
     std::vector<std::pair<std::string, Image>> result;
     for (auto& jsonImage : jsonImages)
     {
-        if (jsonImage.is_object())
+        if (jsonImage.is_object() && jsonImage.contains("path"))
```

Comment:
This change seems a bit random, also should we check for "palette" too?

Reply:
The reason for it, is that `{ "gx": "bla", "y": -1 }` is also an object, but one that should be skipped here. If I don’t add it, it will try to load an empty path as a PNG, fail, and skip loading the object.

(We should not check for `palette` as that parameter is optional.)

Reply:
I see, the usage of getting strings don't really look optional and I looked at `Json::GetString` just now and I see that it has a hidden default parameter, that is why default arguments should be rare it will just end up with situations like this. Thanks for the clarification.